sopel/builtins/xkcd.py

```diff
@@ -124,13 +118,16 @@ def xkcd(bot, trigger):
             if (query.lower() == "latest" or query.lower() == "newest"):
                 requested = latest
             else:
-                number = searchxkcd_search(query)
-                if number is None:
-                    # generic web-search engine as fallback
-                    number = web_search(query)
-                if not number:
-                    bot.reply('Could not find any comics for that query.')
+                try:
+                    number = searchxkcd_search(query)
+                except NoResultsError:
+                    bot.reply("Sorry, I couldn't find any comics for that query.")
                     return
+                except (ResponseFormatError, SearchFailedError, SearchXkcdError):
```

Comment:
Nitpick: Why catch the subclasses here?

Reply:
I'm gonna say it's because I did this at some ungodly wee hour and forgot that they were subclasses despite just creating the new exception types mere moments earlier. Possibly also related to the fact that I started out creating an `Enum` type for the search function to return on errors, before deciding exceptions were better.

tl;dr: I dunno, weird brain stuff. 😹